test.js

```diff
@@ -1,171 +1,245 @@
 import test from 'ava';
 import issueRegex from './index.js';
 
-const matches = [
+const matches = test.macro({
+	exec(t, input, expected, description) {
+		const match = issueRegex().exec(`something ${input} something`);
+		t.truthy(match, `should match but doesn't${description ? `: ${description}` : ''}`);
+
+		t.deepEqual(match.groups, {
+			organization: undefined,
+			repository: undefined,
+			issueNumber: undefined,
+			...expected,
+		}, description);
+
+		// Verify index-based matches
+		t.is(match[1], match.groups.organization);
+		t.is(match[2], match.groups.repository);
+		t.is(match[3], match.groups.issueNumber);
+		t.is(match.length, 4);
+	},
+	title(_, input) {
+		return `should match ${input}`;
+	},
+});
+
+const noMatch = test.macro({
+	exec(t, input, description) {
+		const match = issueRegex().exec(`something ${input} something`);
+		t.falsy(match, description);
+	},
+	title(_, input) {
+		return `should not match ${input}`;
+	},
+});
+
+// Ensure that multiple patterns can be matched at once
+test('baseline', t => {
+	t.deepEqual('Fixes #143 and avajs/ava#1023'.match(issueRegex()), [
+		'#143',
+		'avajs/ava#1023',
+	]);
+});
+
+// Test cases for matching patterns
+test(
+	matches,
 	'#1',
+	{issueNumber: '1'},
+);
+test(
+	matches,
 	'#3223',
+	{issueNumber: '3223'},
+);
+test(
+	matches,
 	'sindresorhus/dofle#33',
+	{organization: 'sindresorhus', repository: 'dofle', issueNumber: '33'},
+);
+test(
+	matches,
 	'foo-bar/unicorn.rainbow#21',
+	{organization: 'foo-bar', repository: 'unicorn.rainbow', issueNumber: '21'},
+);
+test(
+	matches,
 	'foo/a#1',
+	{organization: 'foo', repository: 'a', issueNumber: '1'},
```

Comment:
I tried this object expected values, but it has a high noise-to-signal ratio. I figured arrays would be easier to parse